include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -328,23 +328,6 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
                 BOOST_ASSERT(unpacked_path.size() > 0);
                 unpacked_path.back().turn_instruction = turn_instruction;
                 unpacked_path.back().duration_until_turn += (ed.distance - total_weight);
-
-                if (is_first_segment)
-                {
-                    auto source_weight = start_traversed_in_reverse
-                                             ? phantom_node_pair.source_phantom.reverse_weight
-                                             : phantom_node_pair.source_phantom.forward_weight;
-                    // Given this geometry:
-                    // U---v---w---x---Z
-                    //       s
-                    // The above code will create segments for (v, w), (w,x) and (x, Z).
-                    // However the first segment duration needs to be adjusted to the fact that the
-                    // source phantom is in the middle of the segment.
-                    // We do this by subtracting v--s from the duration.
-
-                    BOOST_ASSERT(unpacked_path.front().duration_until_turn >= source_weight);
-                    unpacked_path.front().duration_until_turn -= source_weight;
-                }
```

Comment:
this code segment is duplicated further down with an additional check. the removal of the check allows for the removal of this if-branch.